cirq/contrib/acquaintance/gates.py

```diff
@@ -0,0 +1,317 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from functools import partial
+from itertools import chain
+from math import ceil
+from operator import indexOf
+from typing import Sequence, Dict, Tuple, List, NamedTuple
+
+from cirq import CompositeGate, TextDiagrammable
+from cirq.ops import (
+        Operation, Gate, gate_features, QubitId, OP_TREE, SWAP, GateOperation)
+from cirq.contrib.acquaintance.shift import CircularShiftGate
+from cirq.contrib.acquaintance.permutation import (
+        PermutationGate, SwapPermutationGate, LinearPermutationGate)
+
+class AcquaintanceOpportunityGate(Gate, TextDiagrammable):
+    """Represents an acquaintance opportunity."""
```

Comment:
Could you describe what an acquaintance opportunity is? E.g. "In a swap network there is a chance for an interaction when two qubits are adjacent; that is an acquaintance opportunity".